protocol/x/subaccounts/keeper/grpc_query_subaccount.go

```diff
@@ -51,13 +51,19 @@ func (k Keeper) Subaccount(
 	}
 	ctx := lib.UnwrapSDKContext(c, types.ModuleName)
 
-	val := k.GetSubaccount(
+	subaccount := k.GetSubaccount(
 		ctx,
 		types.SubaccountId{
 			Owner:  req.Owner,
 			Number: req.Number,
 		},
 	)
-
-	return &types.QuerySubaccountResponse{Subaccount: val}, nil
+	settledSubaccount, _, err := k.GetSettledSubaccount(
```

Comment:
This is still under the `keeper` package, so don't think you need to make `GetSettledSubaccount` public